modules/bvl_feedback/php/module.class.inc

```diff
@@ -35,4 +47,80 @@ class Module extends \Module
     {
         return "Behavioural Feedback";
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param string $type The type of widgets to get
+     * @param \User  $user The user widgets are being retrieved for.
+     *
+     * @return \LORIS\GUI\Widget[]
+     */
+    public function getWidgets(string $type, \User $user)
+    {
+        switch($type) {
+        case 'dashboard':
+            $factory = \NDB_Factory::singleton();
+            $baseURL = $factory->settings()->getBaseURL();
+            $DB      = $factory->database();
+
+            $last_login = $user->getLastLogin($DB);
+
+            $bvl_feedback      = $DB->pselect(
+                "SELECT fbt.Name, fbe.Testdate, fbe.Comment, fbth.FieldName,
+                fbth.CommentID, fbth.SessionID, fbth.CandID, fbth.Feedback_level
+                FROM feedback_bvl_entry fbe
+                JOIN feedback_bvl_thread fbth USING (FeedbackID)
+                JOIN feedback_bvl_type fbt USING (Feedback_type)
+                WHERE fbth.Status='opened' AND fbth.Active='Y'
+                ORDER BY Testdate DESC LIMIT 4",
+                array()
+            );
+            $frontend_feedback = array();
+            foreach ($bvl_feedback as $row) {
+                if (new \DateTime($row['Testdate']) > $last_login) {
+                    $row['new'] = true;
+                } else {
+                    $row['new'] = false;
+                }
+                if ($row['Feedback_level'] === 'profile') {
+                    $row['URL'] = '/' . $row['CandID'];
+                } else if ($row['Feedback_level'] === 'visit') {
+                    $row['URL'] = '/instrument_list/?candID='
+                        . $row['CandID'] . '&sessionID='
+                        . $row['SessionID'];
+                } else if ($row['Feedback_level'] === 'instrument') {
+                    $instrument = $DB->pselectOne(
+                        "SELECT Test_name from flag WHERE CommentID=:cid",
+                        array('cid' => $row['CommentID'])
+                    );
+                    if (!empty($instrument)) {
+                        $row['URL'] = '/instruments/'
+                            . $instrument . '/?candID='
+                            . $row['CandID'] . '&sessionID='
+                            . $row['SessionID'] . '&commentID='
+                            . $row['CommentID'];
+                    }
+                }
+                $frontend_feedback[] = $row;
+            }
+            return [
+                new \LORIS\dashboard\Widget(
+                    new \LORIS\dashboard\WidgetContent(
+                        "Behavioural Feedback Notifications",
+                        $this->renderTemplate(
+                            "dashboardwidget.tpl",
+                            [
+                                'baseURL'       => $baseURL,
+                                'notifications' => $frontend_feedback
+                            ],
```

Comment:
remove " , "

Reply:
Those shouldn't make a difference Travis only runs PHP 7.3+ but I'll remove them and see what happens.